index.src.html

```diff
@@ -849,6 +864,54 @@ <h3 id="workers">Workers</h3>
 
 </section>
 
+<section>
+  <h2 id="premission-prompt">Permission Prompt</h2>
+
+  <em>This section is not normative.</em>
+
+  According to the discutions
+  [[Issue#23](https://github.com/WICG/private-network-access/issues/23)], the
+  <dfn export local-lt="permission prompt">private network access permission
+  prompt</dfn> is introduced to relax mixed-content checks.
+
+  Private network access permission prompt is a <a spec="permissions">permission</a>.
+
+  <h3 id="permission-goals">Goals</h3>
+
+  The overarching goal is to provide a way for private devices which are not able
+  and not welling to directly talk to any public servers including TLS certificate
+  authorities and not able to apply secure contexts to relax mixed-content checks.
+
+  <h3 id="permission-header">Additional CORS Headers</h3>
+
+  The <dfn export http-header>`Private-Network-Access-Name`</dfn> attempt to
+  provide users a human friendly name in the private network access permission
+  prompt.
+
+  <h3 id="permission-states">States</h3>
+
+  The states of the private network access permission can be one of the three:
+
+  * <dfn export>Denied</dfn>:
+
+  * <dfn export>Granted</dfn>:
+
+  * <dfn export>Prompt</dfn>:
+
+  <h3 id="permission-store">Store</h3>
+
+  The user agent maintains a single permission store which is a list of permission
+  store entries. Each particular entry denoted by its descriptor and key can only
+  appear at most once in the list.
+
+  The |key| for the entries should be the IP address of local devices.
+
+  The |discriptor| woule contains the list of origin which gained user permission
+  for the key IP address, the optional name for the entry and the optional id for
+  the entry.
```

Comment:
Should we link these back to the permission [spec](https://w3c.github.io/permissions/) instead of repeating it?

Reply:
+1. In the to-be-written "obtain local network access permission" algorithm, we should probably get and set permission store entries (using the appropriate algorithms from the permissions spec) with the right keys. We can explain the keying schema in plain English for readability too.